sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -133,21 +133,39 @@ def __init__(
         self._populate_attributes(**kwargs)
 
     def __iter__(self):
-        return self
+        return self._iter_contextual_wrapper()
+
+    def _iter_contextual_wrapper(self, max_wait_time=None):
+        original_timeout = None
+        while True:
+            # This is not threadsafe, but gives us a way to handle if someone passes
+            # different max_wait_times to different iterators and uses them in concert.
+            if max_wait_time:
+                original_timeout = self._handler._timeout
+                self._handler._timeout = max_wait_time * 1000
```

Comment:
shouldn't this part be put in front of the while True loop?

otherwise every time `self._handler._timeout` get reset back to max_wait_time in the loop.

Reply:
Yes, that's the intent.  This is part of what lets the nested iterator override behavior work sanely in the tests below.  the iterator ONLY uses the override timeout WHILE DOING ITS ITERATION; so if someone were to do a fetch_next or another iteration within that loop, they'd now control the override flow as if the other loop hadn't changed anything.  (which seems sensible since we're "getting an iter" and that iter may just be overriden, but only for _itself_ and not iters you get afterwards, even if you're not done the first one)

Reply:
thanks for explanation, actually after reading the "respect timeout override" test this part is very clear to me now.
you go deeper than me :)